mono/utils/determine_physical_ram_size.c

```diff
@@ -0,0 +1,73 @@
+/**
+ * Author: Nathan Ricci (naricc@microsoft.com)
+ * Copyright 2019 Microsoft Corp. 
+ */
+
+
+#include <mono/utils/determine_physical_ram_size.h>
+
+#if defined (__APPLE__)
+#include <mach/message.h>
+#include <mach/mach_host.h>
+#include <mach/host_info.h>
+#include <sys/sysctl.h>
+#endif
+#if defined (__NetBSD__)
+#include <sys/param.h>
+#include <sys/sysctl.h>
+#include <sys/vmmeter.h>
+#endif
+
+guint64
+mono_determine_physical_ram_size (void)
+{
+#if defined (TARGET_WIN32)
+	MEMORYSTATUSEX memstat;
+
+	memstat.dwLength = sizeof (memstat);
+	GlobalMemoryStatusEx (&memstat);
```

Comment:
What value does it return when it fails?

Reply:
It return zero in that case. I can make it throw in the case.

Reply:
> I can make it throw

Depending on what you had in mind, code in `utils/` shouldn't assume there's a managed environment.  Normally you can use `MonoError`, but in this case the caller is in sgen, so you have to do Something Else: probably return some sigil value, log an sgen log message and continue.

Reply:
Return MonoError and let sgen translate?

Reply:
@lambdageek I decided to use a log message as you suggested, but I am not sure if I got all the conventions right. I used mono_trace_warning, like so:

mono_trace_warning(MONO_TRACE_GC, "Warning: Unable to determine physical ram size for GCMemoryInfo");
		

Is that good?